pkg/runtime/depot.go

```diff
@@ -45,9 +46,19 @@ type depot struct {
 	fsMutex   *sync.Mutex
 }
 
-func newDepot() (*depot, error) {
+func newDepot(runtimePath string) (*depot, error) {
 	depotPath := filepath.Join(storage.CachePath(), depotName)
 
+	// Windows does not support hard-linking across drives, so determine if the runtime path is on a
+	// separate drive than the default depot path. If so, use a drive-specific depot path.
+	if runtime.GOOS == "windows" {
+		runtimeVolume := filepath.VolumeName(runtimePath)
+		storageVolume := filepath.VolumeName(storage.CachePath())
+		if runtimeVolume != storageVolume {
+			depotPath = filepath.Join(runtimeVolume+"\\", "activestate", depotName)
+		}
+	}
```

Comment:
I can see where you were coming from, it feels awkward dealing with the runtimePath here, since depot shouldn't have any concerns about that. But this might be an ok compromise for now, the only other solution I see is introduce some new construct that is responsible for this information, which feels like overkill for the current use-case.

That said, may be good to highlight this compromise as a comment so we hopefully avoid us further building on this by accident. ie. if we need to extend this logic then we should probably think more about the separate construct.